### DISCUSSION TITLE
More precise (tax) calculations

### DISCUSSION THREAD

**Discussion by bats77** [ghostfolio/ghostfolio#2662] [Ideas]
Hello, I had some suggests to enhance Ghostfolio, already pretty good : 
- total value shall be a field, that will calculate "unit price", otherwise account will be rarely exact ;
- quantity for dividends shall be calculated and prefilled with all (buy and sells) operations before the date ;

It's lacking also for fine tuning about taxes operations : 
- "capital redemption" looks like a dividend, but not taxed as a revenue, because it'is a selling operation only about nominal value - without reducing the quantity of stocks ; so it's a subdivision of "sell" that doesn't affect quantity, just lowering the (bought, historical) value and increasing the future goodwill.
- dividends are published officially, but they often support a pretaxation (usually fixed in percentages) with a tax reclaim (also in percentages) that can be calculated immediately for further reclaim. It's like a second column of fees.

Nice job by the way, pursue 👍

**Comment by pedrommvv** [comment]
`Dividend pretaxation` or `witholding tax` are the same thing right?

Would the community agree to add a `withholding tax` field for dividends? 
Usually brokers or banks do this automatically, retain a percentage for tax purposes.

The `fee` field doesn't fit this requirement.

Awesome project 👏

**Comment by bats77** [comment]
Yes pretaxation or withholding taxe are the same concept.

Fee is for brokers, you could have others fees, as stamp fee (fixed tax) by example on the London Stock Exchange on some sell/buy/merge operations.
Withholding tax, is only on received dividends, calculated in percentages (mainly 10-30% of brutto amount). For example, your bank or broker gives to you an amount in netto (100), so reports (for tax purposes) or yield's performance aren't reliable if based on this, because brutto / real dividend is 130, minus 30 of withholding tax.
If you want to know yield of your stock, you shall calculate it on the real dividend (130 here) as whTax varies about your property (private or business) and your residency towards residency of your equity.

**Reply by pedrommvv** [reply]
This withholding tax feature would be very useful, on dividends, but also interest (from bonds).

**Comment by Karmac** [comment]
This would be a really useful feature for me too. Even if there's no calculations made until future updates, it would be helpful to have a tax input field as soon as posible so we can record the information there. I'm currently using notes for that.

Thanks!

**Comment by samijii** [comment]
I agree with the previous comments about usefulness of having an option to add a "withholding tax" (currency as unit) or a "withholding tax rate" (percentage as unit) in the dividend activity. Withholding tax added by user would be then automatically deducted from the total amount of dividend. By default the withholding tax value should be 0 (no tax deducted). 

Here is a quick proposal how it could look like:
![dividend_withholding_tax](https://github.com/ghostfolio/ghostfolio/assets/11930203/8457df7e-ad23-4e25-8bd8-438cf305a6b1)

P.S. Many thanks to all contributors for an awesome project!

**Reply by fib1985** [reply]
Hi, I agree this would be a great add!

**Comment by dpjrodrigues** [comment]
Other potential option would be to have the tax rate configured at account level (with a toggle on/off). If on, all the transactions that generate profit would automatically populate the respective tax based on the tax rate.

**Reply by SledgehammerPL** [reply]
On the same account there can be different taxes. The tax should be related to asset

**Comment by bats77** [comment]
Yes it could be interesting to have at global level, but it will be insufficient in many cases.
For example, even in a same country, if some common stocks will have a withhold of 15%, on real estate stocks you may have 30-38% of withhold tax, and for "transparent" stocks where stockholder are taxable - in this case there is no withholding tax. So individual setting will fit perfectly.
And by experience, not all moves of a stock are subject to tax, for example when you have "capital redemption", that looks like as a dividend without tax.

**Comment by Torsten-Sc** [comment]
At the end of February, I have usually used up my tax-free allowance and so I really need a view of the tax.
A view of the withholding tax, USA; FR. etc. would also be ideal

**Reply by matzingerhh** [reply]
Same for Germany

**Comment by MiniBlister** [comment]
Agree and would love to see Tax feature implementation

**Comment by NKHnC** [comment]
Implementing Capital Gains Tax calculations for Australian market would be amazing. This would let you compete with the likes of [Sharesight](https://www.sharesight.com/au/) and [Navexa](https://www.navexa.com/au). Here is an open-source project where they did the calculations for CGT, https://github.com/judilsteve/asx-cgt-calculator-optimiser. Doing tax reports would also be extremely helpful! I know this is a lot to ask but thought it worth writing down.

**Reply by Batwam** [reply]
The tax return content is broken down into many categories, cost base adjustments can't be guessed. Wouldn't using ghostfolio for tax return purposes require loading up the end of year reports from Vanguard, iShares,... for each ETFs/Share? 

Seems to me that the best you could get is total dividends and total profit for a given year, and maybe a rough tax estimate if we were able to specify a tax bracket.

**Comment by MiniBlister** [comment]
Any plan to add tax features

**Reply by MiniBlister** [reply]
is this not planned at all?
